src/index.ts

```diff
@@ -151,18 +154,21 @@ async function metadataAccessor<T>(
   }
 
   const requestMethod = fastFail ? fastFailMetadataRequest : request;
-  const res = await requestMethod<T>({
+  const req = {
```

Comment:
[Optional] We may want a type annotation here rather than type assertion (casting):
```suggestion
  const req: GaxiosOptions = {
```

The annotation is slightly safer if this is modified later. Documentation:
- https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions

Reply:
Good call!